### ISSUE TITLE
[plugin-terser] The worker approach to running Terser breaks many valid Terser options

### ISSUE THREAD

**Issue by justinfagnani** [rollup/plugins#1781]
- Rollup Plugin Name: plugin-terser
- Rollup Plugin Version: 0.4.4
- Rollup Version: 4.24.0
- Operating System (or Browser): Linux
- Node Version: v18.20.3
- Link to reproduction _(⚠️ read below)_: https://stackblitz.com/edit/rollup-repro-qx7xkl?file=package.json

### Expected Behavior

All valid Terser options are supported. For this reproduction, the Terser `nth_identifier` option works.

### Actual Behavior

An exception during build because only some of the code for the `nth_identifier` option is serialized. Other things seem to break from serializing the options, like `nameCache` updates.

### Additional Information

The offending line that shows up in stack traces is here: https://github.com/rollup/plugins/blob/4a2358a3f65efc35a6e64bbff00ec5b43de90172/packages/terser/src/worker.ts#L39

But, the technique of taking options in one realm and serializing them to another, like a worker, is just not generally workable. Not all Terser options will be serializable. Functions are an obvious case, but I've had other cases of objects with prototypes, etc., breaking too.

**Comment by midnight-wonderer** [comment]
Wow, that's how `nameCache` doesn't work.

It seems to be officially supported according to the type definition. https://github.com/rollup/plugins/blob/92daef00b0da30de172868d4e0792c8686da0045/packages/terser/src/type.ts#L33

I thought there was something wrong with my setup.
Do you have a workaround?

**Comment by midnight-wonderer** [comment]
Found one!

Basically, `nameCache` is supported. However, the issue lies in the parallel worker execution.  
On each successful terser execution, the `nameCache` output gets merged into the central cache object.  

The problem arises when the same name appears on multiple files that are being processed in parallel. Each mangles independently, sending out conflicting output cache.

The workaround is setting `maxWorkers` to 1.

**Comment by justinfagnani** [comment]
That workaround won't work for `nth_identifier` unfortunately, since it can be a function and functions can't be reliably serialized.
